text_quality/language/fasttext.py

```diff
@@ -0,0 +1,124 @@
+        super().__init__()
+
+        if not model_file.exists():
+            self._download_model(model_file)
+        self._model = fasttext.load_model(str(model_file))
+
+        self._line_threshold = line_threshold
+
+    def classify(self, text: str) -> tuple[str, float]:
+        """Classify a text string.
+
+        Args:
+            text (str): The text to classify.
+        Returns:
+            A tuple[str, float] with the language code (e.g. "nl") and the confidence.
+        """
+
+        lines: list[str] = [
+            LanguageClassifier.preprocess(line) for line in text.split("\n")
+        ]
+        line_labels, line_confidences = self._model.predict(
+            lines, threshold=self._line_threshold
+        )
+        language, confidence = FastTextLanguageClassifier._aggregate_lines(
+            line_labels, line_confidences
+        )
+
+        return language.removeprefix(self._LABEL_PREFIX), confidence
+
+    @staticmethod
+    def _download_model(model_file: Path):
+        try:
+            url = FastTextLanguageClassifier.MODEL_URLS[model_file.name]
+        except KeyError as e:
+            raise ValueError(
+                f"No download location known for '{model_file.name}'."
+            ) from e
+
+        model_file.parent.mkdir(parents=True, exist_ok=True)
+
+        logging.info("Downloading from '%s', storing at '%s' .", model_file, url)
+        with urllib.request.urlopen(url) as response, model_file.open("wb") as out_file:
+            shutil.copyfileobj(response, out_file)
+
+    @staticmethod
+    def _aggregate_lines(
+        line_labels: list[list[str]], line_confidences: list[ArrayLike]
+    ) -> tuple[str, float]:
+        """Aggregate the results per line from the classifier.
+
+        Args:
+            line_labels (list[list[str]]): the labels returned by the classifier; one list of labels for each input line
+            line_confidences (list[ArrayLike]): the confidences returned by the classifier; one array for each input line
+        Returns:
+            A tuple[str, float] with the language code (e.g. "nl") and the confidence.
+                ("None", 0.0) if no input line could be classified.
+        """
+        if len(line_labels) != len(line_confidences):
+            raise ValueError(
+                f"Labels and confidences must have the same length, but were {len(line_labels)} and {len(line_confidences)}."
+            )
+
+        if any(labels for labels in line_labels):
+            total_confidences = Counter()
+            for labels, confidences in zip(line_labels, line_confidences):
+                # Iterate over labels and confidences per input line
+                # Number of entries per line corresponds to k parameter in model.predict(), defaults to 1
+                for label, confidence in zip(labels, confidences):
+                    total_confidences[label] += confidence
+            label, total_confidence = total_confidences.most_common()[0]
```

Comment:
```suggestion
            label, total_confidence = total_confidences.most_common(1)
```
also works 🙃

Reply:
Not as far as I can see, because `most_common(1)` still returns a list of tuples, even though it contains only a single element:

```python
In [9]: total_confidences.most_common(1)
Out[9]: [(`__label__nl`, 0.8)]

In [10]: label, confidence = total_confidences.most_common(1)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[10], line 1
----> 1 label, confidence = total_confidences.most_common(1)

ValueError: not enough values to unpack (expected 2, got 1)

```
This works, but does not really help:
```python
In [11]: label, confidence = total_confidences.most_common(1)[0]

In [12]: label
Out[12]: __label__nl

In [13]: confidence
Out[13]: 0.8
```